net-reporter-api/src/api/total_http_requests/total_http_requests_filters.rs

```diff
@@ -0,0 +1,400 @@
+use ion_rs;
+
+use ion_rs::element::reader::ElementReader;
+use ion_rs::IonReader;
+use ion_rs::IonType;
+use ion_rs::IonWriter;
+use ion_rs::ReaderBuilder;
+
+use ion_rs::StreamItem;
+use net_core_api::api::API;
+use net_core_api::encoder_api::Encoder;
+use net_core_api::decoder_api::Decoder;
+use net_core_api::typed_api::Typed;
+
+const DATA_TYPE: &str = "total_http_requests_filters";
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct TotalHttpRequestsFiltersDTO {
+    http_methods: Vec<String>,
+    include_http_methods_mode: Option<bool>,
+    endpoints: Vec<String>,
+    include_endpoints_mode: Option<bool>,
+    bytes_lower_bound: Option<i64>,
+    bytes_upper_bound: Option<i64>,
+}
+impl API for TotalHttpRequestsFiltersDTO { }
+
+impl TotalHttpRequestsFiltersDTO {
+    pub fn new(
+        http_methods: &[String],
+        include_http_methods_mode: Option<bool>,
+        endpoints: &[String],
+        include_endpoints_mode: Option<bool>,
+        bytes_lower_bound: Option<i64>,
+        bytes_upper_bound: Option<i64>,
+    ) -> Self {
+        TotalHttpRequestsFiltersDTO {
+            http_methods: http_methods.to_vec(),
+            include_http_methods_mode,
+            endpoints: endpoints.to_vec(),
+            include_endpoints_mode,
+            bytes_lower_bound,
+            bytes_upper_bound,
+        }
+    }
+
+    pub fn get_http_methods(&self) -> &[String] {
+        self.http_methods.as_slice()
+    }
+
+    pub fn is_include_http_methods_mode(&self) -> Option<bool> {
```

Comment:
There might be a conversation about if we should name returning `Option<T>` methods `is_..`